core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -30,7 +30,7 @@ class InterBrokerSendThread(name: String,
                             networkClient: NetworkClient,
                             requestGenerator: () => Iterable[RequestAndCompletionHandler],
                             time: Time)
-  extends ShutdownableThread(name, isInterruptible = false) {
+  extends ShutdownableThread(name, isInterruptible = true) {
```

Comment:
@guozhangwang Thanks for the review. I have made `isInterruptible` a parameter so that it can be overridden if required.